src/Workspaces/Core/Portable/Remote/RemoteHostClient.cs

```diff
@@ -220,7 +215,7 @@ protected Connection()
             public abstract Task<T> InvokeAsync<T>(string targetName, IReadOnlyList<object> arguments, CancellationToken cancellationToken);
             public abstract Task<T> InvokeAsync<T>(string targetName, IReadOnlyList<object> arguments, Func<Stream, CancellationToken, Task<T>> funcWithDirectStreamAsync, CancellationToken cancellationToken);
 
-            protected virtual void Dispose(bool disposing)
+            protected virtual void DisposeImpl()
```

Comment:
💭 Even this one seems like it would benefit from just using `Dispose()`

Reply:
We have a pattern where the base class only calls DisposeImpl if the object has not been disposed yet.